plumbing/objectid.go

```diff
@@ -62,11 +61,38 @@ func FromBytes(in []byte) (ObjectID, bool) {
 	return id, true
 }
 
+func NewObjectIDFromHex(in string) ObjectID {
+	f := format.SHA1
+	if len(in) == SHA256HexSize {
+		f = format.SHA256
+	}
+
+	v, _ := hex.DecodeString(in)
+	id := ObjectID{
+		format: f,
+	}
+	copy(id.hash[:], v)
+	return id
+}
+
+func NewObjectIDFromBytes(in []byte) ObjectID {
+	f := format.SHA1
+	if len(in) == sha256.Size {
+		f = format.SHA256
+	}
+
+	id := ObjectID{
+		format: f,
+	}
+	copy(id.hash[:], in)
+	return id
+}
+
```

Comment:
I removed this on [this commit](https://github.com/pjbgf/go-git/commit/b4db92e80aa69f21417a0cbb42ae1f2ba5d05b48). This is no longer needed.

```suggestion
```